src/NumberTheory/QuadFormAndIsom/finite_group_actions.jl

```diff
@@ -0,0 +1,871 @@
+  )
+  T = orthogonal_submodule(L, S)
+  @req is_definite(T) "Orthogonal complement is not definite"
+
+  if rank(T) == 0
+    P = OS
+  else
+    OT = isometry_group(T; kwargs...)
+    gen = _stabilizer_in_diagonal_action(L, S, T, OS, OT; gen_fun)
+    P = matrix_group(gen)
+  end
+  return P
+end
+
+@doc raw"""
+    stabilizer_sublattice_in_orthogonal_group(
+      L::ZZLat,
+      S::Union{QQMatrix, ZZLat};
+      stable::Bool=false,
+      special::Bool=false,
+      check::Bool=true,
+      kwargs...,
+    ) -> MatrixGroup
+
+Given a definite sublattice ``S`` of an even lattice ``L``, or a generating
+set of vectors given as rows in a matrix with rational entries, return the
+stabilizer of ``S`` in the orthogonal group of ``L``.
+
+Note that the function requires that ``S`` and its orthogonal complement ``T``
+in ``L`` are both definite. However ``S`` need not be primitive in ``L``.
+
+If `stable` is set to `true`, return the largest subgroup of the stable
+orthogonal group $O^\#(L)$ preserving ``S``. If `special` is set
+to `true`, return the largest subgroup of the special orthogonal group
+$SO(L)$ preserving ``S``. If both keyword arguments are `true`,
+then return the intersection of the corresponding stabilizers.
+
+If `check` is set to `true`, the function tests whether ``S`` is a sublattice
+of ``L``, and whether ``S`` and ``T`` are definite.
+
+The keyword arguments in `kwargs` are optional arguments for the computation
+of isometry group of definite lattices (see [`isometry_group(::ZZLat)`](@ref)).
+
+# Examples
+```jldoctest
+julia> A2 = root_lattice(:A, 2)
+Integer lattice of rank 2 and degree 2
+with gram matrix
+[ 2   -1]
+[-1    2]
+
+julia> L = lattice_in_same_ambient_space(A2, QQ[1 1; 1 -1])
+Integer lattice of rank 2 and degree 2
+with gram matrix
+[2   0]
+[0   6]
+
+julia> H1 = stabilizer_sublattice_in_orthogonal_group(A2, L)
+Matrix group of degree 2
+  over rational field
+
+julia> describe(H1)
+"C2 x C2"
+
+julia> H2 = stabilizer_sublattice_in_orthogonal_group(A2, L; stable=true, special=true)
+Matrix group of degree 2
+  over rational field
+
+julia> order(H2)
+1
+```
+"""
+function stabilizer_sublattice_in_orthogonal_group(
+    L::ZZLat,
+    S::ZZLat;
+    stable::Bool=false,
+    special::Bool=false,
+    check::Bool=true,
+    kwargs...,
+  )
+  if check
+    @req ambient_space(L) === ambient_space(S) "Lattices are not contained in the same quadratic space"
+    @req is_sublattice(L, S) "Matrix does not define a sublattice"
+    @req is_definite(S) "The lattice S is not definite"
+  end
+  # Compute the isometries preserving the rational span of S
+  # If S is not saturated, we then filter the ones actually preserving S 
+  if !is_primitive(L, S)
+    is_sat = false
+    Ssat = primitive_closure(L, S)
+  else
+    is_sat = true
+    Ssat = S
+  end
+  OSsat = orthogonal_group(Ssat; kwargs...)
+
+  P = _maximal_extension(L, Ssat, OSsat; kwargs...)
+  if !is_sat
+    P, _ = stabilizer(P, S, on_lattices)
```

Comment:
```suggestion
    P, _ = stabilizer(P, S, on_lattices)
```
Perhaps it would be faster to go via the discriminant groups? 
How do these compare in terms of speed?